File: majorProject/src/main/java/com/aocc/majorproject/ui/UiBounds.java
package com.aocc.majorproject.ui;

import com.aocc.framework.Input;

/** Axis-aligned rectangle in world coordinates (1280×720). */
public final class UiBounds {

    public final int x;
    public final int y;
    public final int width;
    public final int height;

    public UiBounds(int x, int y, int width, int height) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }

    public int centerX() {
        return x + width / 2;
    }

    public int centerY() {
        return y + height / 2;
    }

    public boolean contains(float touchX, float touchY) {
        return touchX > x && touchX < x + width - 1
                && touchY > y && touchY < y + height - 1;
    }

    public boolean contains(Input.TouchEvent event) {
        return contains(event.x, event.y);
    }

    /** Returns a smaller rectangle inset on all sides (for visual focus rings). */
    public UiBounds inset(int horizontal, int vertical) {
        return new UiBounds(x + horizontal, y + vertical,
                width - horizontal * 2, height - vertical * 2);
    }
}

File: majorProject/src/main/kotlin/com/aocc/majorproject/ui/UiBounds.kt
package com.aocc.majorproject.ui

import com.aocc.framework.Input

/** Axis-aligned rectangle in world coordinates (1280×720). */
data class UiBounds(
    @JvmField val x: Int,
    @JvmField val y: Int,
    @JvmField val width: Int,
    @JvmField val height: Int,
) {

    fun centerX(): Int = x + width / 2

    fun centerY(): Int = y + height / 2

    fun contains(touchX: Float, touchY: Float): Boolean {
        return touchX > x && touchX < x + width - 1
            && touchY > y && touchY < y + height - 1
    }

    fun contains(event: Input.TouchEvent): Boolean {
        return contains(event.x.toFloat(), event.y.toFloat())
    }

    /** Returns a smaller rectangle inset on all sides (for visual focus rings). */
    fun inset(horizontal: Int, vertical: Int): UiBounds {
        return UiBounds(
            x + horizontal,
            y + vertical,
            width - horizontal * 2,
            height - vertical * 2
        )
    }
}
